### PR TITLE
[textarea] Add adaptToContent and maxRows properties

### PR DESCRIPTION
**Pull request by meluru**
Another approach, much better imho

### CODE REVIEW

**Review by dparret — APPROVED** [review]
Works well locally.

File: pyrene/src/components/TextArea/TextArea.tsx [hunk 1]
```diff
@@ -89,10 +101,39 @@ const TextArea: React.FC<TextAreaProps> = ({
...
   const characterCount = maxLength - (value !== null ? value.length : 0);
   const characterLimitReached = characterCount < 0;
+  const textAreaRef = useRef<HTMLTextAreaElement>(null);
+  const [currentRows, setCurrentRows] = useState(minRows);
+  const [text, setText] = useState(value);
+
+  useEffect(() => {
+    if (adaptToContent && textAreaRef.current) {
+      // eslint-disable-next-line no-bitwise
+      const rowsFromHeight = ~~(textAreaRef.current.scrollHeight / textAreaLineHeight);
```
Comment (sanmiawi) [inline pyrene/src/components/TextArea/TextArea.tsx:119]: I think `Math.floor()` is easier to read and `~~` starts to fail when dividing very large numbers.

Comment (meluru) [reply pyrene/src/components/TextArea/TextArea.tsx:119]: That makes sense, thanks :) Will fix it

File: pyrene/src/components/TextArea/TextArea.tsx [hunk 2]
```diff
@@ -89,10 +101,39 @@ const TextArea: React.FC<TextAreaProps> = ({
...
+  const textAreaLineHeight = 16;
   const characterCount = maxLength - (value !== null ? value.length : 0);
   const characterLimitReached = characterCount < 0;
+  const textAreaRef = useRef<HTMLTextAreaElement>(null);
+  const [currentRows, setCurrentRows] = useState(minRows);
+  const [text, setText] = useState(value);
+
+  useEffect(() => {
+    if (adaptToContent && textAreaRef.current) {
+      // eslint-disable-next-line no-bitwise
```
Comment (sanmiawi) [inline pyrene/src/components/TextArea/TextArea.tsx:118]: I guess this can also be removed this since we are no longer doing the bitwise operation

**Pull request merged**